dspy/streaming/streaming_listener.py

```diff
@@ -51,6 +52,9 @@ def __init__(
         self.chat_adapter_start_identifier = f"[[ ## {self.signature_field_name} ## ]]"
         self.chat_adapter_end_identifier = re.compile(r"\[\[ ## (\w+) ## \]\]")
 
+        self.xml_adapter_start_identifier = f"<{self.signature_field_name}>"
+        self.xml_adapter_end_identifier = re.compile(rf"</{self.signature_field_name}>")
+
```

Comment:
Since we already have three adapters, shall we define a mapping for adapter-specific parameters? Alternatively, we can move the identifier into each adapter class for better encapsulation